samza-test/src/test/java/org/apache/samza/test/framework/StreamApplicationIntegrationTestHarness.java

```diff
@@ -229,18 +232,32 @@ public Config getConfig() {
   }
 
   @Override
-  protected KafkaProducer createProducer() {
-    Properties kafkaConfig = new Properties();
-    kafkaConfig.setProperty("bootstrap.servers", bootstrapServers());
-    return new KafkaProducer<>(kafkaConfig, STRING_SERIALIZER, STRING_SERIALIZER);
+  protected Properties createProducerConfigs() {
+    Properties producerProps = new Properties();
+    producerProps.setProperty(CommonClientConfigs.BOOTSTRAP_SERVERS_CONFIG, bootstrapServers());
+    /*
+     * Stream application tests uses string serde for both key and value.
+     * The default serde for the test producer in IntegrationTestHarness uses
+     * byte serde for value and string for key.
+     */
+    producerProps.setProperty(KEY_SERIALIZER_CLASS_CONFIG, STRING_SERIALIZER);
```

Comment:
Could you implement this by calling `super.createProducerConfigs`, and then just override the serializers? Same for `createConsumerConfigs`.